src/components/listings/ImageGrid.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react'
+import { View } from 'react-native'
+import ImageGridItem from '@/components/listings/ImageGridItem'
+import { ImageGridProps } from '@/types/imageGrid'
+
+function ImageGrid(props: ImageGridProps) {
+  const { images, onSelectedImagesChange } = props
+
+  const [selectedImages, setSelectedImages] = useState<Array<number>>(
+    props.selectedImages || [],
+  )
+
+  function handleSelectedImages(images: Array<number>) {
+    setSelectedImages(images)
+    onSelectedImagesChange(selectedImages)
```

Comment:
use `images` here instead of `selectedImages` as the setState is asynchronous.
it might not have been updated by the time this line fires